docs/test_plan.md

```diff
@@ -0,0 +1,51 @@
+# Mbed OS Tools Test Plan
+
+The goal of delivering the unified Mbed OS tools is to improve testing and quality. We plan to deliver this by doing the following:
+
+1. Preserve existing tests while migrating from the individual repositories to the unified repository
+1. Begin gathering metrics in the continuous integration system to measure the initial state of the code and tests, as well as to measure future improvements
+1. Bring up code coverage to a reasonable level
+1. For core sections of the codebase, write [Black-box test cases](https://en.wikipedia.org/wiki/Black-box_testing) to enforce the current behavior (but remain agnostic to the implementation).
+1. Begin refactoring the codebase to decrease duplication, remove unsupported APIs, and to set us up for future feature development
+1. Establish a public facing (and documented) API
+1. When the API has stabilized, release 1.0 and deprecate the legacy tools packages
+
+## Existing test suite
+
+The tools are currently in separate packages:
+
+- `mbed-ls` (Mbed LS): https://github.com/ARMmbed/mbed-ls
+- `mbed-host-tests` (htrun): https://github.com/ARMmbed/htrun
+- `mbed-greentea` (greentea): https://github.com/ARMmbed/greentea
+
+They all have their own test suites of varying completeness. Part of the effort of bringing these projects into the unified tools packages was to get these test suites passing consistently under all operating systems.
+
+Once the test suites were all passing, their codebases were migrated to this repository (preserving the history through [git subtrees](https://git-scm.com/book/en/v1/Git-Tools-Subtree-Merging)). From their, the modules were moved to their current locations and the test suites were merged. The combined test suite is now ran on all PRs submitted to this repository in our continuous integration system.
```

Comment:
"From there?" or "From their"? - Apart from that this looks great, Excellent :-)

Reply:
Oops yup, good catch!

Reply:
Updated!